apps/web/server/lib/router/getServerSideProps.ts

```diff
@@ -87,6 +61,32 @@ export const getServerSideProps = async function getServerSideProps(
       },
     },
   });
+}
+
+export const getServerSideProps = async function getServerSideProps(context: GetServerSidePropsContext) {
+  const queryParsed = querySchema.safeParse(context.query);
+  const pathWithQuery = context.req.url || "";
+  const onlyPath = pathWithQuery.split("?")[0];
+  const isEmbed = onlyPath.endsWith("/embed");
+  const pageProps = {
+    isEmbed,
+  };
+
+  if (!queryParsed.success) {
+    log.warn("Error parsing query", queryParsed.error);
+    return {
+      notFound: true,
+    };
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { form: formId, embed, ...fieldsResponses } = queryParsed.data;
+  const { currentOrgDomain } = orgDomainConfig(context.req);
+
+  let timeTaken: Record<string, number | null> = {};
```

Comment:
Commented at other places as well but comment here as well. Time Taken is an existing approach that I am adding more data on. This later needs to move to Sentry performance tracing